web/datapipeline/sap_systems_projector.go

```diff
@@ -79,31 +81,43 @@ func SAPSystemsProjector_SAPSystemsDiscoveryHandler(dataCollectedEvent *DataColl
 			instance.SystemReplicationStatus = parseReplicationStatus(i.SystemReplication)
 			addSAPControlData(&instance, i.SAPControl)
 
-			err := storeSAPInstance(db,
-				instance,
-				"id", "sid", "type", "features", "instance_number",
-				"system_replication", "system_replication_status",
-				"sap_hostname", "start_priority", "http_port", "https_port", "status",
-				"tenants", "db_host", "db_name")
+			instances = append(instances, instance)
+		}
+
+		err := storeSAPInstances(db,
+			instances,
+			"id", "sid", "type", "features", "instance_number",
+			"system_replication", "system_replication_status",
+			"sap_hostname", "start_priority", "http_port", "https_port", "status",
+			"tenants", "db_host", "db_name")
+		if err != nil {
+			return err
+		}
 
+		if len(instances) > 0 {
+			for _, instance := range instances {
+				db = db.Where("NOT (id = ? AND agent_id = ? AND instance_number = ?)", s.Id, dataCollectedEvent.AgentID, instance.InstanceNumber)
```

Comment:
It looks like we still have some remaining `Id` instead of `ID`...
It would be totally the same, but maybe `... instance.ID, instance.AgentID, instance.InstanceNumber)` looks better